client/src/fabric/FabricClientConnection.ts

```diff
@@ -53,7 +53,7 @@ export class FabricClientConnection extends FabricConnection implements IFabricC
         await this.connectInner(connectionProfile, certificate, privateKey);
     }
 
-    getConnectionDetails(): any {
+    async getConnectionDetails(): Promise<any> {
```

Comment:
you could set the type to be `{connectionProfile: any, certificatePath: string, privateKeyPath: string}`